### PR TITLE
Necessary respec changes

### PR DESCRIPTION
**Pull request by iherman**
This provides all the changes to respec to change the document's short name.

Merge should happen only once the repository also changes, to avoid respec and subsequent pubrules' checkeer errors.

@deniak I have one question: in https://github.com/w3c/vc-jwt/pull/115#issuecomment-1620408736 we said that a change should happen manually on the document before the publication step, namely (quoting from the pubrules document):

> The syntax of a “history” URI must be https://www.w3.org/standards/history/shortname/, and consistent with the shortname mentioned in 'Latest Version'. Note: If there's a shortname change it must be specified using the following data attribute data-previous-shortname='previous-shortname' on the <a> element.

However, the WD generated by respec does not include and 'history' reference, so I am not sure where I would have to make a change. Please advise, thx.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jwt/pull/126.html" title="Last updated on Jul 13, 2023, 10:10 AM UTC (b7cc3d5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jwt/126/227499a...b7cc3d5.html" title="Last updated on Jul 13, 2023, 10:10 AM UTC (b7cc3d5)">Diff</a>

### CODE REVIEW

**Comment by OR13** [comment]
next step is rename the repo, then ping me to merge this.

**Comment by OR13** [comment]
@iherman this was what I had manually generated previously, for reference... https://github.com/w3c/vc-jwt/pull/125/commits/c6b17eef668bf195ac37d670d52355ed7a606121

**Comment by iherman** [comment]
> @iherman this was what I had manually generated previously, for reference... [c6b17ee](https://github.com/w3c/vc-jwt/commit/c6b17eef668bf195ac37d670d52355ed7a606121)
> 
> I ran the spec locally exported it manually, and edited the file.

Yes, that makes absolute sense, but I would need the thumbs up from @deniak to see whether this is really what the process document means by 'history' field.

**Comment by iherman** [comment]
@OR13 I have renamed the repository. However, before merging, I would like to get an approval from @deniak (which won't happen before Monday; France is on public holiday today).

**Comment by OR13** [comment]
Awesome, thank you!

**Review by selfissued — APPROVED** [review]

**Review by mprorock — APPROVED** [review]

**Comment by deniak** [comment]
I understand the shortname is going from 'vc-jwt' to 'vc-jose-cose'. If so, the commit https://github.com/w3c/vc-jose-cose/commit/c6b17eef668bf195ac37d670d52355ed7a606121 looks good.
Updating the shortname in the respec config should leave the `data-previous-shortname` attribute to 'vc-jwt' which is required so the system will know there is a shortname change.

**Comment by iherman** [comment]
> I understand the shortname is going from 'vc-jwt' to 'vc-jose-cose'. If so, the commit [c6b17ee](https://github.com/w3c/vc-jose-cose/commit/c6b17eef668bf195ac37d670d52355ed7a606121) looks good. Updating the shortname in the respec config should leave the `data-previous-shortname` attribute to 'vc-jwt' which is required so the system will know there is a shortname change.

I do not understand, @deniak. If I look at the [preview](https://pr-preview.s3.amazonaws.com/w3c/vc-jwt/pull/126.html), which is the pure HTML generated by respec, there is no trace of that `data-previous-shortname` attribute. I have also tried to add the 

```json
"previousDiffURI": "https://www.w3.org/TR/2023/WD-vc-jwt-20230710/"
```

entry to the respec config (which is the only entry I could find that refers to previous short name), the entry is not there either.

**Comment by iherman** [comment]
> > I do not understand, @deniak. If I look at the [preview](https://pr-preview.s3.amazonaws.com/w3c/vc-jwt/pull/126.html), which is the pure HTML generated by respec, there is no trace of that `data-previous-shortname` attribute.
> 
> I think there's a bug in respec. There should be a link to the history page in the history section of the headers. This is where the data attribute should appear.

That is our luck :-) Does it take a long time to get this done in your estimation? (I do not know who maintains respec these days). Alternatively, can you tell us what is what you expect exactly in the header? We can then add that manually and we can move on.

(The pubrules' reference does not help either: at the moment, the only "history" reference is the github commit history...)

**Comment by deniak** [comment]
respec does check if [the history link exists](https://github.com/w3c/respec/blob/main/src/w3c/headers.js#L763) before displaying it. I'm not sure why the check is there but that's why your spec is missing the link, because of course, the link with the new shortname doesn't exist yet.

What you can do is generating the snapshot from respec and then add the history link manually, e.g. https://www.w3.org/TR/rdf-canon/:
```html
<dt>History:</dt>
<dd>
    <a href="https://www.w3.org/standards/history/vc-jose-cose/" data-previous-shortname="vc-jwt">https://www.w3.org/standards/history/vc-jose-cose/</a>
</dd>
<dd>
    <a href="https://github.com/w3c/vc-jose-cose/commits/">Commit history</a>
</dd>
```

Once the link is there, you can upload it on /TR and it should pass pubrules.

After the document has been published, the history link should magically appear for future versions as the history link will return a 200.

**Comment by deniak** [comment]
Bug reported: https://github.com/w3c/respec/issues/4486

**Comment by OR13** [comment]
Can we merge this PR now? or do we have to wait to resolve the bug?

**Comment by iherman** [comment]
FYI unexpected hiccup: it seems that there is, formally, a need to get an official approval for a short name change. I have submitted this request yesterday but, with vacations and all, it may take more and the official publication may not happen tomorrow (Thursday 20th).

The possible changes to change a date are on me. In the meantime, it is important to _keep all PR-s open_ to avoid getting into a versioning mess.

Cc @OR13

**Comment by iherman** [comment]
The document has been published: https://www.w3.org/TR/2023/WD-vc-jose-cose-20230720/

After merging this branch, and _before_ merging the other pending PRs:

- The echidna script must be updated at https://github.com/w3c/vc-jose-cose/blob/55c1d0d38a9fd495b71d008b32a0f9b7aefb2b38/.github/workflows/auto-publish.yml.off#L19 The short name should be updated
- The script must be 're-instated'

**Comment by OR13** [comment]
done in https://github.com/w3c/vc-jose-cose/pull/129

**Pull request merged**
